### PR TITLE
Improved steps ignoring in Recorder extension with meta steps and annotations support.

### PR DESCRIPTION
**Pull request by sspat**
Implementation for proposed feature described in #5209

### CODE REVIEW

**Comment by DavertMik** [comment]
Sorry, #5221 was bigger and I had to merge it first.
Could you try to solve the merge conflict and update this PR?

**Comment by sspat** [comment]
Done!

File: ext/Recorder.php [hunk 1]
```diff
@@ -555,6 +560,8 @@ public function afterStep(StepEvent $e)
     }
 
     /**
+     * @param Step $step
+     *
      * @return bool
      */
     protected function isStepIgnored(StepEvent $e)
```
[the remark this answers is not in the document]
Comment (sspat) [reply ext/Recorder.php]: Sorry, missed that one during conflict resolution! Thanks for the review!

File: ext/Recorder.php [hunk 2]
```diff
@@ -35,7 +35,10 @@
  * * `delete_successful` (default: true) - delete screenshots for successfully passed tests  (i.e. log only failed and errored tests).
  * * `module` (default: WebDriver) - which module for screenshots to use. Set `AngularJS` if you want to use it with AngularJS module. Generally, the module should implement `Codeception\Lib\Interfaces\ScreenshotSaver` interface.
  * * `ignore_steps` (default: []) - array of step names that should not be recorded, * wildcards supported. Meta steps can also be ignored.
```
[the remark this answers is not in the document]
Comment (sspat) [reply ext/Recorder.php]: I agree with you, this was again my inattention during the merge, it's back now

**Comment by OneEyedSpaceFish** [comment]
Seems good to me, what do you reckon @DavertMik ?

**Pull request merged**